paimon-core/src/main/java/org/apache/paimon/operation/KeyValueFileStoreScan.java

```diff
@@ -99,6 +100,11 @@ public KeyValueFileStoreScan(
         this.mergeEngine = mergeEngine;
         this.changelogProducer = changelogProducer;
         this.fileIndexReadEnabled = fileIndexReadEnabled;
+        this.onlyReadRealBuckets = onlyReadRealBuckets;
+
+        if (onlyReadRealBuckets) {
+            super.withBucketFilter(bucket -> bucket >= 0);
```

Comment:
this will overwrite old bucket filter, or be overwrited new bucket filter, maybe we should add a method to skip negative buckets...